nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -140,38 +127,59 @@ define(['StagingServiceClient', 'common/html', 'common/runtime', 'common/ui', 'u
             this.cssBaseClass = cssBaseClass;
         }
 
+        _setUpModel() {
+            // fill in the options for the types
+            const typeArray = Object.keys(this.model.getItem('params'));
+            xsvAppSpec.parameters[0].dropdown_options.options = typeArray.sort().map((opt) => {
+                return {
+                    display: this.fileTypeMapping[opt],
+                    value: opt,
+                };
+            });
+            xsvAppSpec.parameters[0].default_values = typeArray;
+            this.xsvAppSpec = xsvAppSpec;
+            this.spec = Spec.make({ appSpec: this.xsvAppSpec });
+
+            const internalModelContext = {
+                app: {
+                    spec: this.spec,
+                },
+                params: this.spec.makeDefaultedModel(),
+                inputs: {
+                    otherParamIds: this.xsvAppSpec.parameters.map((param) => {
+                        return param.id;
+                    }),
+                },
+            };
```

Comment:
This is the model for the xsv template generator "app" -- it's used by the `paramsWidget` module to generate and validate the inputs on the XSV template form.